### PR TITLE
Add typing for steps/execute/upgrade.py

### PR DESCRIPTION
**Pull request by janhavlin**
Blocked until #1328 is merged.

### CODE REVIEW

**Review by happz — APPROVED** [review]

File: tmt/steps/execute/upgrade.py [hunk 1]
```diff
@@ -106,42 +106,46 @@ class ExecuteUpgrade(ExecuteInternal):
...
+            return cast(tmt.steps.discover.Discover, self.step.plan.discover)
 
-    def go(self, guest):
+    def go(self, guest: tmt.steps.provision.Guest) -> None:
         """ Execute available tests """
-        self._results = []
+        self._results: List[tmt.base.Result] = []
         # Inform about the how, skip the actual execution
-        super(ExecutePlugin, self).go(guest)
+        ExecutePlugin.go(self, guest)
```
Comment (psss) [inline tmt/steps/execute/upgrade.py:148]: What's the motivation for this change?

**Review by psss — COMMENTED** [review]
Looks good, just one question.

File: tmt/steps/execute/upgrade.py (hunk 1 above)
Comment (janhavlin) [reply tmt/steps/execute/upgrade.py:148]: If I remember it correctly, we intend to run `ExecutePlugin`'s `go()` method. The previous code, `super(ExecutePlugin, self).go(guest)` calls the `go()` method of `ExecutePlugin`'s parent, which is `Plugin`. Does that sound correct?

Comment (psss) [reply tmt/steps/execute/upgrade.py:148]: I thought, and that's what the comment above suggests, that we just want to show info about `how` which is handled in the parent. @FrNecas, any suggestion here?

Comment (janhavlin) [reply tmt/steps/execute/upgrade.py:148]: We already discussed this so I tried to dig into IRC to refresh my memory. The issue was that mypy complained about the original code:
```
tmt/steps/execute/[upgrade.py:135](http://upgrade.py:135/): error: Argument 2 for "super" not an instance of argument 1
```

The thought process on how to fix the above issue went like this:

* fnecas (IRC): Hm, what about ExecutePlugin.go(self, guest)?
* mprchlik (IRC): that would be acceptable, actually, but it's inconsistent with the use of super(). on the other hand, the whole point is inconsistent with regular super() use, so yeah, +1
and we'd need no ignore rule for mypy, big fat comment still applies
* fnecas (IRC): possibly also super(ExecuteInternal, self).go(guest) would work but IMO it's more unclear what it actually does
* Jan Havlin: fnecas (IRC): doesn't super(ExecutePlugin, self) mean the parent of ExecutePlugin? so an instance of Plugin?
* Jan Havlin: fnecas (IRC): then it would be Plugin.go(self, guest)
* fnecas (IRC): Jan Havlin: Yeah, it does, good point, I probably got confused while implementing it :) That's why I'd rather be explicit in ExecutePlugin.go(self, guest) to avoid this confusion. I think it's desired that ExecutePlugin.go() is called
* fnecas (IRC): because the upgrade plugin can use --exit-first from ExecutePlugin.options() as well
* fnecas (IRC): and ExecutePlugin.go() just calls Plugin.go() and informs about exit-first option value (which is obviously nice for the upgrade plugin too)

Comment (psss) [reply tmt/steps/execute/upgrade.py:148]: Understood. Thanks much for the clarification!

**Review by psss — APPROVED** [review]
